tests/extmod/vfs_posix_ilistdir_filter.py

```diff
@@ -0,0 +1,45 @@
+# Test ilistdir filter of . and .. for VfsPosix.
+
+try:
+    import os
+
+    os.VfsPosix
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+
+def test(testdir):
+    vfs = os.VfsPosix(testdir)
+
+    expected = {".a": True, "..a": True, "...a": True, "a.": True}
+    actual = {}
+
+    for dir in expected:
+        vfs.mkdir(dir)
+
+    for entry in vfs.ilistdir("/"):
+        actual[entry[0]] = True
+
+    assert expected == actual
+
+
+# We need an empty directory for testing.
+# Skip the test if it already exists.
```

Comment:
Do we want to skip the test, or should we just delete the directory and then create it again? Or fail the test?

Reply:
This behaviour I got from vfs_posix_ilistdir_del.py.  I can delete the directory and recreate it if that is preferred.

Reply:
If you are following a pattern that is already used in tests here, then I think we should leave it as-is unless we have a good reason to change it.